src/test/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSparkUnitTest.java

```diff
@@ -0,0 +1,312 @@
+        final File expectedFilterPassingFile = new File(TEST_DATA_DIR, filenamePrefix + ".pp.ec.filter.pass.fa");
+        final String filterPassingFileName = compareNamesAndComputeSeqEditDist(actualFilterPassingFile, expectedFilterPassingFile, true, editDistancesBetweenSeq);
+        for(final Integer d : editDistancesBetweenSeq){ Assert.assertEquals(d, zero); }
+
+        editDistancesBetweenSeq = new ArrayList<>();
+        final File filterPassingFile = new File(workingDir, filterPassingFileName);
+        final File actualRmdupFile = new File(workingDir, RunSGAViaProcessBuilderOnSpark.runSGARmDuplicate(sgaPath, filterPassingFile, workingDir, indexer, indexerArgs, runtimeInfo));
+        final File expectedRmdupFile = new File(TEST_DATA_DIR, filenamePrefix + ".pp.ec.filter.pass.rmdup.fa");
+        final String duplicateRemovedFileName = compareNamesAndComputeSeqEditDist(actualRmdupFile, expectedRmdupFile, false, editDistancesBetweenSeq);
+        for(final Integer d : editDistancesBetweenSeq){ Assert.assertEquals(d, zero); }
+
+        editDistancesBetweenSeq = new ArrayList<>();
+        final File duplicateRemovedFile = new File(workingDir, duplicateRemovedFileName);
+        final File actualMergedFile = new File(workingDir, RunSGAViaProcessBuilderOnSpark.runSGAFMMerge(sgaPath, duplicateRemovedFile, workingDir, indexer, indexerArgs, runtimeInfo));
+        final File expectedMergedFile = new File(TEST_DATA_DIR, filenamePrefix + ".pp.ec.filter.pass.rmdup.merged.fa");
+        final String mergedFileName = compareNamesAndComputeSeqEditDist(actualMergedFile, expectedMergedFile, false, editDistancesBetweenSeq);
+        for(final Integer d : editDistancesBetweenSeq){ Assert.assertEquals(d, zero); }
+
+        // final assembled contig test
+        final File mergedFile = new File(workingDir, mergedFileName);
+        final File actualAssembledContigsFile = new File(workingDir, RunSGAViaProcessBuilderOnSpark.runSGAOverlapAndAssemble(sgaPath, mergedFile, workingDir, runtimeInfo));
+
+        compareContigs(actualAssembledContigsFile, expectedAssembledContigFile);
+    }
+
+    private static void compareContigs(final File actualAssembledContigFile,
+                                       final File expectedAssembledContigFile)
+            throws IOException, InterruptedException{
+
+        final List<String> actualFASTAContents   = (null==actualAssembledContigFile  ) ? null : Files.readAllLines(Paths.get(actualAssembledContigFile.getAbsolutePath()  ));
+        final List<String> expectedFASTAContents = (null==expectedAssembledContigFile) ? null : Files.readAllLines(Paths.get(expectedAssembledContigFile.getAbsolutePath()));
+
+        // first create mapping from read length to sequence
+        // (bad, but contig names are not guaranteed to be reproducible in different runs)
+        final SortedMap<Integer, String> actualMap = collectContigsByLength(actualFASTAContents);
```

Comment:
\[FindBugs\] WARNING: NP: Null passed for nonnull parameter of collectContigsByLength\(List\) in org\.broadinstitute\.hellbender\.tools\.spark\.sv\.RunSGAViaProcessBuilderOnSparkUnitTest\.compareContigs\(File, File\)